src/main/java/edu/rice/pcdp/runtime/BaseTask.java

```diff
@@ -86,18 +79,13 @@ public void compute() {
             }
         }
 
-        @Override
-        public void onCompletion(final CountedCompleter<?> caller) {
-            countDownLatch.countDown();
-        }
-
         /**
          * Wait for all tasks registered on this finish scope to complete.
          */
         public void awaitCompletion() {
             try {
-                countDownLatch.await();
-            } catch (final InterruptedException ex) {
+                join();
```

Comment:
Can we add a comment here stating that the `join` allows us to gracefully handle exceptions while computing the `runnable`?